vispy/gloo/buffer.py

```diff
@@ -125,9 +125,8 @@ def resize_bytes(self, size):
         self._glir.command('SIZE', self._id, size)
         # Invalidate any view on this buffer
         for view in self._views:
-            if view() is not None:
-                view()._valid = False
-        self._views = []
+            view._valid = False
```

Comment:
So iterating over the `WeakSet` automatically dereferences them? (no need for the parentheses)

Reply:
Yep!  That's exactly correct. And you're guaranteed to never get a bad reference, since you always get the dereferenced object.